weave/server.py

```diff
@@ -66,6 +67,10 @@ class HandleRequestResponse:
 def handle_request(
     request, deref=False, serialize_fn=storage.to_python
 ) -> HandleRequestResponse:
+    if not cache.is_current_cache_valid():
+        logging.info("Invalidating cache")
+        cache.set_cache_timestamp()
```

Comment:
hmm why do we pass the cache information via the env var vs a context variable? Was the context missing setting the cache in other places? 

This is not safe because another request can cause the cache to be updated while one is already in flight.